recipes-docs/rzg2l-readme/files/README.md

```diff
@@ -0,0 +1,1845 @@
+```
+
+### MIPI CSI2 with Arducam 5MP MIPI Camera
+
+RZG2L-SBC supports the MIPI CSI-2 camera interface and the Arducam 5MP MIPI Camera (OV5640 image sensor) is enabled and tested.
+
+You should edit `uEnv.txt` as follows to enable MIPI CSI-2 interface with the camera supported:
+
+```
+enable_overlay_csi_ov5640=1
+```
+To use the camera, we need to enable the CSI-2 module. Run the following commands:
+
+```
+root@rzg2l-sbc:~# cd /home/root/
+root@rzg2l-sbc:~# ./v4l2-init.sh <resolution>
+```
+
+The <resolution> argument specifies the resolution for the camera. Valid resolutions are:
+
+- 1280x720
+- 1280x960
+- 1600x900
+- 1920x1080
+- 1920x1200
+- 2560x1080
+
+If no resolution is specified or an invalid resolution is provided, the default resolution 1280x960 will be used. For example:
+
+When use a valid resolution:
+
+```
+root@rzg2l-sbc:~# ./v4l2-init.sh 1920x1080
+Link CRU/CSI2 to ov5640 1-003c with format UYVY8_2X8 and resolution 1920x1080
+```
+
+When no resolution is specified:
+
+```
+root@rzg2l-sbc:~# ./v4l2-init.sh
+No resolution specified. Using default resolution: 1280x960
+Link CRU/CSI2 to ov5640 1-003c with format UYVY8_2X8 and resolution 1280x960
+```
+
+When an invalid resolution is provided:
+
+```
+root@rzg2l-sbc:~# ./v4l2-init.sh 3000x2000
+Invalid resolution: 3000x2000
+Input resolution is not available. Using default resolution: 1280x960
+Link CRU/CSI2 to ov5640 1-003c with format UYVY8_2X8 and resolution 1280x960
+```
+
+The `v4l2-init.sh` script helps to enable the CSI-2 module and select supported display resolution for the camera as well.
+
+Run the following to open Camera and preview the video on the screen.
+
+```
+root@rzg2l-sbc:~# gst-launch-1.0 v4l2src device=/dev/video0 ! videoconvert ! waylandsink
+```
+
+### Generic USB Bluetooth framework
+
+The RZG2L-SBC supports the generic USB Bluetooth framework, which is back-ported from the Linux kernel mainline. TP-Link UB500 Bluetooth 5.0 Nano USB Adapter (Realtek chipset) has been tested and proven to work on the board.
+
+The following steps will guide how to enable the TP-Link UB500 adapter:
+
+- Step 1: Download the appropriate firmware for the TP-Link UB500 adapter and store it on the RZG2L-SBC. This will ensure it is loaded each time the board boots (one-time setup).
+
+```shell
+root@rzg2l-sbc:~# mkdir -p /lib/firmware/rtl_bt
+root@rzg2l-sbc:~# curl -s https://raw.githubusercontent.com/Realtek-OpenSource/android_hardware_realtek/rtk1395/bt/rtkbt/Firmware/BT/rtl8761b_fw -o /lib/firmware/rtl_bt/rtl8761bu_fw.bin
+```
+**Note:**
+**(1) Please make sure you have internet access before running the commands.**
+
+**(2) If the firmware is being downloaded for the first time, a reboot of the board is required to ensure the TP-Link UB500 adapter functions properly.**
+
+- Step 2: Verify whether the TP-Link UB500 adapter is properly attached.
+
+Run the following command to ensure that the system has recognized the TP-Link UB500 adapter:
+
+```shell
```

Comment:
@nam-vo-khac-rvc ,

Currently, the Bluetooth feature is blocked by software by default. Could you please update this document with steps to unblock and up then Bluetooth